pallets/gear/src/lib.rs

```diff
@@ -1436,14 +1443,14 @@ pub mod pallet {
 
                 Self::deposit_event(event);
             } else {
-                // Message in mailbox is not meant for any processing, hence 0 gas limit
-                // and no gas tree needs to be created
-                let origin = who.into_origin();
                 let message = message.into_stored(ProgramId::from_origin(origin));
 
-                // TODO: update logic of insertion into mailbox following new
-                // flow and deposit appropriate event (issue #1010).
-                MailboxOf::<T>::insert(message.clone())?;
+                if gas_limit >= T::MessageRent::get() {
+                    // TODO: update logic of insertion into mailbox following new
+                    // flow and deposit appropriate event (issue #1010).
```

Comment:
If we insert message in mailbox, than we need to deposit this event:
```rust
Pallet::<T>::deposit_event(Event::UserMessageSent {
    message,
    expiration: Some(T::BlockNumber::zero()),
});
```
T::BlockNumber::zero() will be replaced with the real one soon

if we don't insert message into mailbox, but for showing it in events, we should deposit:
```rust
Pallet::<T>::deposit_event(Event::UserMessageSent {
    message,
    expiration: None,
});
```

Apply this logic everywhere, please

Reply:
Also for removing everywhere.
If you want, you can separate this cutting logic into different PR and then come back to this one.